Goal: Re-activate ECHOCTL in login

Fixes: #429758

Status wrt upstream: Not reported yet. 

Note: Was removed by upstream in 4.0.8 with "remove dead code" comment

Index: shadow-4.0.18.1/src/login.c
===================================================================
--- shadow-4.0.18.1.orig/src/login.c	2007-06-22 19:14:00.712717643 +0200
+++ shadow-4.0.18.1/src/login.c	2007-06-22 19:17:24.702331390 +0200
@@ -175,6 +175,16 @@
 	termio.c_lflag |= ISIG | ICANON | ECHO | ECHOE;
 	termio.c_iflag |= ICRNL;
 
+#if defined(ECHOKE) && defined(ECHOCTL)
+	termio.c_lflag |= ECHOKE | ECHOCTL;
+#endif
+#if defined(ECHOPRT) && defined(NOFLSH) && defined(TOSTOP)
+	termio.c_lflag &= ~(ECHOPRT | NOFLSH | TOSTOP);
+#endif
+#ifdef ONLCR
+	termio.c_oflag |= ONLCR;
+#endif
+
 	/* leave these values unchanged if not specified in login.defs */
 	termio.c_cc[VERASE] = getdef_num ("ERASECHAR", termio.c_cc[VERASE]);
 	termio.c_cc[VKILL] = getdef_num ("KILLCHAR", termio.c_cc[VKILL]);
